src/common/RefCountedObj.cc

```diff
@@ -0,0 +1,40 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+//
+#include "include/ceph_assert.h"
+
+#include "common/RefCountedObj.h"
+#include "common/ceph_context.h"
+#include "common/dout.h"
+#include "common/valgrind.h"
+
+RefCountedObject::~RefCountedObject()
+{
+  ceph_assert(nref == 0);
+}
+
+void RefCountedObject::put() const {
+  CephContext *local_cct = cct;
+  auto v = --nref;
```

Comment:
Should probably make this an int so we aren't doing a bunch of atomic dereferences on that lsubdout.

Reply:
The type of `v` is `long unsigned int` for me. The `++` prefix/postfix operator returns the type of the atomic.